docs/pages/deploy-a-cluster/high-availability.mdx

```diff
@@ -0,0 +1,369 @@
+events. For etcd, Teleport uses namespaces within item keys to identify audit
+events and cluster state data.
+
+The Teleport Auth Service manages the creation of any required DynamoDB tables
+or Firestore collections itself, and does not require them to exist in advance.
+
+<Admonition title="Required permissions">
+
+In your cloud provider's RBAC solution (e.g., AWS or Google Cloud IAM), your
+Teleport instances need permissions to read from and write to your chosen
+key/value store, as well as to create tables and collections (if your key/value
+store supports them).
+
+</Admonition>
+
+## Object storage service
+
+High-availability Teleport deployments use an object storage service for
+persisting session recordings. The Teleport Auth Service supports two object
+storage services:
+
+- Amazon S3 
+- Google Cloud Storage
+
+In your Teleport configuration (described in the [Configuration](#configuration)
+section), you must name a Google Cloud Storage or Amazon S3 bucket to use for
+managing session recordings. The Teleport Auth Service creates this bucket, so
+to prevent unexpected behavior, you should not create it in advance. 
+
+<Admonition title="Required permissions">
+
+In your cloud provider's RBAC solution, your Teleport instances need
+permissions to create, read from, and write to buckets.
+
+</Admonition>
+
+## TLS credential provisioning 
+
+High-availability Teleport deployments require an automated system to fetch
+certificates from Let's Encrypt and provision Teleport instances with them. 
+
+If you are running a single instance of the Teleport Auth Service and Proxy
+Service, you can configure this instance to fetch credentials for itself from
+Let's Encrypt using the [ACME ALPN-01
+challenge](https://letsencrypt.org/docs/challenge-types/#tls-alpn-01), where
+Teleport demonstrates that it controls the ALPN server at the HTTPS address of
+your Teleport Proxy Service.
+
+For high-availability deployments, however, it is much more efficient to use a
+separate system to request TLS credentials from Let's Encrypt and provision
+Teleport instances with them. In this case, with multiple Teleport instances
+behind a load balancer, you will need to use the [ACME
+DNS-01](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge)
+challenge to demonstrate domain name ownership to Let's Encrypt. In this
+challenge, your TLS credential provisioning system creates a DNS TXT record with
+a value expected by Let's Encrypt.
+
+In the configuration we are demonstrating in this guide, each Teleport instance
+expects TLS credentials for HTTPS to be available at the file paths
+`/etc/teleport-tls/tls.key` (private key) and `/etc/teleport-tls/tls.crt`
+(certificate).
+
+## DNS service
+
+Set up a DNS zone where you can create records for Teleport, e.g., an Amazon
+Route 53 hosted zone or Google Cloud DNS zone.
+
+You must create a DNS A record that points to the address of your load balancer. 
```

Comment:
small Nit but in AWS if assigning a load balancer as the destination IIRC it uses an Alias and that is more like a CNAME record than A record.  Stating you must create an A record could create confusion.

Reply:
Thanks! What DNS records do HA setups usually include on AWS, Google Cloud, and on-prem deployments?

Reply:
In pretty much every case the minimum needed is either an A record or CNAME from `teleport.example.com` to the public load balancer IP or DNS name in front of the Teleport proxies.

(AWS' Route 53 also has the concept of aliases which are basically an internally-expanded CNAME with optional health checks; for all intents and purposes the outcome is the same as a CNAME to the public load balancer DNS name)

When using application access we also encourage people to add the same for `*.teleport.example.com`